mmr/mmr.go

```diff
@@ -479,13 +491,14 @@ func (m *Proof) calculatePeaksHashes(leaves []types.Leaf, mmrSize uint64, proofs
 
 	peaks := GetPeaks(mmrSize)
 	var peaksHashes [][]byte
-	for _, peaksPos := range peaks {
-		lvs := takeWhileVec(&leaves, func(l types.Leaf) bool {
-			return LeafIndexToPos(l.Index) <= peaksPos
+	for _, peakPos := range peaks {
```

Comment:
as a general performance rule, would be more efficient to re-write all the `for loop` that are written with `range` to standard for loop impl. This is because every loop with `range` creates a copy of the var in memory and since we are trying to extract higher benchmarks would make sense to avoid doing copies and thus improve the performance a bit.

Reply:
https://blog.fearcat.in/a?ID=00001-6e405919-74eb-41fe-b22a-fdc8a0cae817

Reply:
makes sense.